### PR TITLE
Remove the insanity around OrganizationFactory usages

### PR DESCRIPTION
**Pull request by i386**
# Description

See [JENKINS-XXXXX](https://issues.jenkins-ci.org/browse/JENKINS-XXXXX).

# Submitter checklist
- [ ] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

File: blueocean-rest/src/main/java/io/jenkins/blueocean/rest/model/BluePipeline.java [hunk 1]
```diff
@@ -11,6 +11,7 @@
 import org.kohsuke.stapler.json.JsonBody;
```
Comment (i386) [inline blueocean-rest/src/main/java/io/jenkins/blueocean/rest/model/BluePipeline.java]: @vivek this was the major change. Most places in the codebase that need `BlueOrganization` references already have a `BluePipeline` in scope.

**Comment by i386** [comment]
@vivek low priority PR.

**Comment by i386** [comment]
@alvarolobato @vivek ping - please review.

File: blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/MatrixProjectImpl.java [hunk 2]
```diff
@@ -29,32 +31,32 @@
...
+        super(organization, folder, parent);
         this.matrixProject = folder;
     }
 
     @Extension(ordinal = 1)
     public static class PipelineFactoryImpl extends BluePipelineFactory{
 
         @Override
-        public MatrixProjectImpl getPipeline(Item item, Reachable parent) {
+        public MatrixProjectImpl getPipeline(Item item, Reachable parent, BlueOrganization organization) {
```
Comment (alvarolobato) [inline blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/MatrixProjectImpl.java]: Something I've found is that in many cases the `Reachable parent` it is already the organization and that we may just change the type or check if it is an organization. Un less we want to support hierarchies in the future.
Have you thought about it? Did I get it wrong?

See: https://github.com/jenkinsci/blueocean-plugin/blob/master/blueocean-pipeline-scm-api/src/main/java/io/jenkins/blueocean/scm/api/AbstractMultiBranchCreateRequest.java#L61

File: blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/MultiBranchPipelineContainerImpl.java [hunk 3]
```diff
@@ -29,9 +34,9 @@ public MultiBranchPipelineContainerImpl(OrganizationFolder folder, Reachable par
     public BlueMultiBranchPipeline get(String s) {
         MultiBranchProject mbp =  folder.getItem(s);
         if(mbp == null){
-            throw new ServiceException.NotFoundException("Multibranch pipeline "+ s + " not found.");
```
Comment (alvarolobato) [inline blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/MultiBranchPipelineContainerImpl.java]: How is this related to this PR?

File: blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/OrganizationFolderPipelineImpl.java [hunk 4]
```diff
@@ -93,8 +96,14 @@ public BlueRunContainer getRuns() {
     }
 
     @Override
-    public String getOrganization() {
-        return pipelineFolder.getOrganization();
+    public String getOrganizationName() {
+        return pipelineFolder.getOrganizationName();
```
Comment (alvarolobato) [inline blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/OrganizationFolderPipelineImpl.java]: Shouldn't this be just `organization.getName()`?

File: blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/MultiBranchPipelineContainerImpl.java (hunk 3 above)
Comment (i386) [inline blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/MultiBranchPipelineContainerImpl.java]: Small clean up - returning null will produce the 404.

File: blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/OrganizationFolderPipelineImpl.java (hunk 4 above)
Comment (i386) [inline blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/OrganizationFolderPipelineImpl.java]: Fixed

File: blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/MatrixProjectImpl.java (hunk 2 above)
Comment (i386) [inline blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/MatrixProjectImpl.java]: but sometimes parent is an Folder that does not implement organization

File: blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/QueueItemImpl.java [hunk 5]
```diff
@@ -28,15 +28,18 @@
...
+        this(organization,
+            item,
             pipeline.getName(),expectedBuildNumber,
             pipeline.getQueue().getLink().rel(Long.toString(item.getId())),
             pipeline.getLink());
     }
 
-    QueueItemImpl(Queue.Item item, String name, int expectedBuildNumber, Link self, Link parent) {
+    QueueItemImpl(BlueOrganization organization, Queue.Item item, String name, int expectedBuildNumber, Link self, Link parent) {
+        this.organization = organization;
```
Comment (i386) [inline blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/QueueItemImpl.java]: Fixed

File: blueocean-pipeline-scm-api/src/main/java/io/jenkins/blueocean/scm/api/AbstractPipelineCreateRequest.java [hunk 6]
```diff
@@ -72,4 +72,8 @@ protected ModifiableTopLevelItemGroup getParent() {
 
         return parent;
     }
+
+    protected BlueOrganization findOrganization() {
+        return OrganizationFactory.getInstance().get(getOrganization());
```
Comment (i386) [inline blueocean-pipeline-scm-api/src/main/java/io/jenkins/blueocean/scm/api/AbstractPipelineCreateRequest.java]: @alvarolobato this code was written by @vivek - I am not sure if this is correct or not.

**Comment by i386** [comment]
@alvarolobato https://github.com/jenkinsci/blueocean-plugin/pull/1304

File: blueocean-pipeline-scm-api/src/main/java/io/jenkins/blueocean/scm/api/AbstractPipelineCreateRequest.java (hunk 6 above)
Comment (vivek) [inline blueocean-pipeline-scm-api/src/main/java/io/jenkins/blueocean/scm/api/AbstractPipelineCreateRequest.java]: @alvarolobato This won't work with multi-org case. What we need is to update [BluePipelineCreateRequest.create()](https://github.com/jenkinsci/blueocean-plugin/blob/8d8d5234579167ebbc072afda09d554d002ec8b0/blueocean-rest/src/main/java/io/jenkins/blueocean/rest/model/BluePipelineCreateRequest.java#L41) and pass BlueOrganization parameter. Then pass organization when calling create: https://github.com/jenkinsci/blueocean-plugin/blob/4cefa8d4a94bb513d252e436ae8228f809368209/blueocean-rest/src/main/java/io/jenkins/blueocean/rest/model/BluePipelineContainer.java#L54.

Comment (alvarolobato) [inline blueocean-pipeline-scm-api/src/main/java/io/jenkins/blueocean/scm/api/AbstractPipelineCreateRequest.java]: @vivek agreed. I think we can address this when removing the "get first organization" from the preloaders problem.
